examples/step-33/step-33.cc

```diff
@@ -1765,37 +1772,54 @@ namespace Step33
     typedef Sacado::Fad::DFad<double> FluxMatrix[EulerEquations<dim>::n_components][dim];
     FluxMatrix *flux = new FluxMatrix[n_q_points];
 
+    typedef double FluxMatrixOld[EulerEquations<dim>::n_components][dim];
+    FluxMatrixOld  *flux_old = new FluxMatrixOld[n_q_points];
+
     typedef Sacado::Fad::DFad<double> ForcingVector[EulerEquations<dim>::n_components];
     ForcingVector *forcing = new ForcingVector[n_q_points];
 
+    typedef double ForcingVectorOld[EulerEquations<dim>::n_components];
+    ForcingVectorOld *forcing_old = new ForcingVectorOld[n_q_points];
+
```

Comment:
The context is this:
```
    typedef Sacado::Fad::DFad<double> ForcingVector[EulerEquations<dim>::n_components];
    ForcingVector *forcing = new ForcingVector[n_q_points];

    for (unsigned int q=0; q<n_q_points; ++q)
      {
        EulerEquations<dim>::compute_flux_matrix (W_theta[q], flux[q]);
        EulerEquations<dim>::compute_forcing_vector (W_theta[q], forcing[q]);
```
where
```
    template <typename InputVector, typename number>
    static
    void compute_forcing_vector (const InputVector &W,
                                 number (&forcing)[n_components])
```
In other words, we pass around a statically sized array. I actually like this much better than having a dynamically sized `std::vector`. 

I could imagine that something like this could work:
```
    typedef std::array<Sacado::Fad::DFad<double>,EulerEquations<dim>::n_components> ForcingVector;
    std::vector<ForcingVector> forcing (n_q_points);

    for (unsigned int q=0; q<n_q_points; ++q)
      {
        EulerEquations<dim>::compute_flux_matrix (W_theta[q], flux[q]);
        EulerEquations<dim>::compute_forcing_vector (W_theta[q], forcing[q]);

```
In that case, `compute_forcing_vector` needs to accept a `std::array<T,N>` as argument instead of the reference to an array.